src/fondant/component/data_io.py

```diff
@@ -170,79 +157,46 @@ def write_dataframe(
         dataframe: dd.DataFrame,
         dask_client: t.Optional[Client] = None,
     ) -> None:
-        write_tasks = []
+        columns_to_produce = [
+            column_name for column_name, field in self.component_spec.produces.items()
+        ]
 
-        dataframe.index = dataframe.index.rename("id")
```

Comment:
We are assuming that the index is called "id" in the loader, so I think we still need this?